source/features/restore-file.tsx

```diff
@@ -42,23 +42,34 @@ async function getFile(filePath: string): Promise<{isTruncated: boolean; text: s
 	return repository.file;
 }
 
-async function dropFile(progress: (message: string) => void, menuItem: Element, filePath: string): Promise<void> {
+async function discardChanges(progress: (message: string) => void, filePath: string): Promise<void> {
 	const file = await getFile(filePath);
 
-	if (!file) {
-		// The file was created by this PR.
-		// This code won’t be reached if `highlight-deleted-and-added-files-in-diffs` works.
-		throw new Error('Nothing to drop. Delete file instead');
+	if (file?.isTruncated) {
+		throw new Error('File too big, you’ll have to use git');
 	}
 
-	if (file.isTruncated) {
-		throw new Error('Drop failed: File too big');
-	}
+	// Only possible if `highlight-deleted-and-added-files-in-diffs` is broken or disabled
+	const isNewFile = !file;
+
+	const change = isNewFile ? `
+		deletions: [
+			{
+				path: "${filePath}"
+			}
+		]
+	` : `
+		additions: [
+			{
+				path: "${filePath}",
+				contents: "${btoa(unescape(encodeURIComponent(file.text)))}"
+			}
+		]
+	`;
 
 	const {nameWithOwner, branch: prBranch} = getBranches().head;
-	progress(menuItem.closest('[data-file-deleted="true"]') ? 'Undeleting…' : 'Committing…');
+	progress('Committing…');
```

Comment:
It's not necessary to differentiate the status here. In every case we're "Committing…"